src/components/ItemsList.js

```diff
@@ -101,82 +103,104 @@ const ItemsList = () => {
-          <button type="submit" onClick={handleRedirect}>
-            Add First Item
-          </button>
-        </div>
-      )}
-      {listHasAtLeastOneItem && (
-        <div className="search-list">
-          <SearchBar
-            value={searchTerm}
-            setValue={(searchTerm) => {
-              setSearchTerm(searchTerm);
-            }}
-          />
-          <ul>
-            {shoppingList[0].items
-              .filter((shoppingItemObject) =>
-                shoppingItemObject.shoppingListItemName
-                  .toLowerCase()
-                  .includes(searchTerm.toLowerCase()),
-              )
-              .sort(sortShoppingList)
-              .map((shoppingItemObject, index) => {
-                const shopIndex = shoppingList[0].items.indexOf(
-                  shoppingItemObject,
-                );
-                return (
-                  <li
-                    key={shoppingItemObject.shoppingListItemName + index}
-                    style={{
-                      backgroundColor: getShoppingItemBackgroundStyles(
-                        shoppingItemObject.daysLeftForNextPurchase,
-                        getDaysBetweenCurrentAndPreviousPurchase,
-                        shoppingItemObject.lastPurchasedOn,
-                      ),
-                    }}
-                  >
-                    <input
-                      type="checkbox"
-                      id={shoppingItemObject.shoppingListItemName}
-                      onChange={() => markItemAsPurchased(shopIndex)}
-                      checked={wasItemPurchasedWithinLastOneDay(
-                        shoppingItemObject.lastPurchasedOn,
-                      )}
-                    />
-                    <label
-                      htmlFor={shoppingItemObject.shoppingListItemName}
-                      aria-label={getItemDescription(
-                        shoppingItemObject.daysLeftForNextPurchase,
-                      )}
-                    >
-                      {shoppingItemObject.shoppingListItemName}
-                    </label>
-                    <button
-                      className="delete-button"
-                      onClick={() =>
-                        deleteItemFromShoppingList(
-                          shoppingItemObject.shoppingListItemName,
-                        )
-                      }
+    <div className="max-h-screen flex flex-col box-border items-center">
+      <header className="bg-green-400 w-full fixed text-center">
+        <h2 className="pt-6 pb-16 text-4xl font-thin text-gray-100">
+          Your Shopping List
+        </h2>
+        <span className="text-white top-0 right-0 absolute sm:mt-4 sm:mr-4">
+          <HomeIcon />
+        </span>
+      </header>
+      <main className="bg-white relative w-full h-full mt-24 rounded-t-3xl overflow-auto">
+        {loading && (
+          <img className="m-auto w-12" src={spinner} alt="Loading..." />
+        )}
```

Comment:
Nice! I think taking this logic outside the return statement and doing like a conditional rendering to return the spinner if loading is true or return the rest of the compoenent when its over. That way the nav wont show during the loading state and after nav renders with the data fetched from db. What do you guys think?

Reply:
Yeah, sounds good to me.